src/lib/statistics/statistics_objects/min_max_filter.hpp

```diff
@@ -11,19 +11,18 @@
 namespace opossum {
 
 /**
- *  Filter that stores a segment's minimum and maximum value
+ * Filter that stores a minimum and maximum value for a number of rows.
+ * MinMaxFilters are typically created for a single chunk and can be used to check whether
+ * a certain value exists in the segment.
 */
 template <typename T>
+// TODO @Bouncner: Could you clarify the "Filter" terminology? What makes a filter a filter? Histograms can be used
+// for pruning, too.
+// TODO @Bouncner: Why do we have MinMaxFilter/RangeFilter if we could also create a histogram? Are these cheaper to
```

Comment:
Why are they "much smaller" than a histogram with a single bin? Couldn't we add a shortcut to the creation of a single bin histogram so that it also simply looks for min / max values?

Reply:
Ah ok, you talk about single bin histograms. That should barely make a difference. We might lose a little bit of performance when we need to prune on columns with many bins (let's assume the column is worth having a large histogram) due to worse caching. But I doubt it would make a real difference.
We can probably merge them, yes. For range filters, that's a different story since they optimize to recognize prunable ranges and not for accuracy as histograms do.

Reply:
> We can probably merge them, yes. For range filters, that's a different story since they optimize to recognize prunable ranges and not for accuracy as histograms do.

That's only a question of how the histograms are created.

I believe that it is possible but probably not worth the effort and might introduce some confusion (see previous comment).

Reply:
We actually thought quite a bit about that during @timzimmermann (may god forgive his moral flaws) and while it sounds rather simple, it can become quickly a nightmare when one tries to balance pruning and accuracy.

But that's just a side note. Full ack to your comment.